## ET Entrega 1 
# Caso Análisis de Condiciones Climáticas en Australia

* Maura Ruminot
* Paula Romero

## Contexto

<font color = 'blue'>
EDA: 

1. [Carga y Revisión de Data](#1)
2. [Descripción de variables](#2)
1. [Análisis de variables](#3)
    * [Variables Categóricas](#4)
    * [Variables Numéricas](#5)
1. [Análisis con gráficos](#6)
1. [ ](#7)

<a id = "1"></a><br>
# Carga y Revisión de Data

In [14]:
# Importación de librerías

import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns
from collections import Counter
import warnings
warnings.filterwarnings("ignore")

In [15]:
data = pd.read_csv('weatherAUS.csv')
data.head()

Date Location  MinTemp  MaxTemp  Rainfall  Evaporation  Sunshine  \
0  2008-12-01   Albury     13.4     22.9       0.6          NaN       NaN   
1  2008-12-02   Albury      7.4     25.1       0.0          NaN       NaN   
2  2008-12-03   Albury     12.9     25.7       0.0          NaN       NaN   
3  2008-12-04   Albury      9.2     28.0       0.0          NaN       NaN   
4  2008-12-05   Albury     17.5     32.3       1.0          NaN       NaN   

  WindGustDir  WindGustSpeed WindDir9am  ... Humidity3pm  Pressure9am  \
0           W           44.0          W  ...        22.0       1007.7   
1         WNW           44.0        NNW  ...        25.0       1010.6   
2         WSW           46.0          W  ...        30.0       1007.6   
3          NE           24.0         SE  ...        16.0       1017.6   
4           W           41.0        ENE  ...        33.0       1010.8   

   Pressure3pm  Cloud9am  Cloud3pm  Temp9am  Temp3pm  RainToday  RISK_MM  \
0       1007.1       8.0       NaN     16.9     21.8         No      0.0   
1       1007.8       NaN       NaN     17.2     24.3         No      0.0   
2       1008.7       NaN       2.0     21.0     23.2         No      0.0   
3       1012.8       NaN       NaN     18.1     26.5         No      1.0   
4       1006.0       7.0       8.0     17.8     29.7         No      0.2   

   RainTomorrow  
0            No  
1            No  
2            No  
3            No  
4            No  

[5 rows x 24 columns]

### * Información de la data

In [16]:
data.describe()

MinTemp        MaxTemp       Rainfall   Evaporation  \
count  141556.000000  141871.000000  140787.000000  81350.000000   
mean       12.186400      23.226784       2.349974      5.469824   
std         6.403283       7.117618       8.465173      4.188537   
min        -8.500000      -4.800000       0.000000      0.000000   
25%         7.600000      17.900000       0.000000      2.600000   
50%        12.000000      22.600000       0.000000      4.800000   
75%        16.800000      28.200000       0.800000      7.400000   
max        33.900000      48.100000     371.000000    145.000000   

           Sunshine  WindGustSpeed   WindSpeed9am   WindSpeed3pm  \
count  74377.000000  132923.000000  140845.000000  139563.000000   
mean       7.624853      39.984292      14.001988      18.637576   
std        3.781525      13.588801       8.893337       8.803345   
min        0.000000       6.000000       0.000000       0.000000   
25%        4.900000      31.000000       7.000000      13.000000   
50%        8.500000      39.000000      13.000000      19.000000   
75%       10.600000      48.000000      19.000000      24.000000   
max       14.500000     135.000000     130.000000      87.000000   

         Humidity9am    Humidity3pm    Pressure9am    Pressure3pm  \
count  140419.000000  138583.000000  128179.000000  128212.000000   
mean       68.843810      51.482606    1017.653758    1015.258204   
std        19.051293      20.797772       7.105476       7.036677   
min         0.000000       0.000000     980.500000     977.100000   
25%        57.000000      37.000000    1012.900000    1010.400000   
50%        70.000000      52.000000    1017.600000    1015.200000   
75%        83.000000      66.000000    1022.400000    1020.000000   
max       100.000000     100.000000    1041.000000    1039.600000   

           Cloud9am      Cloud3pm        Temp9am        Temp3pm        RISK_MM  
count  88536.000000  85099.000000  141289.000000  139467.000000  142193.000000  
mean       4.437189      4.503167      16.987509      21.687235       2.360682  
std        2.887016      2.720633       6.492838       6.937594       8.477969  
min        0.000000      0.000000      -7.200000      -5.400000       0.000000  
25%        1.000000      2.000000      12.300000      16.600000       0.000000  
50%        5.000000      5.000000      16.700000      21.100000       0.000000  
75%        7.000000      7.000000      21.600000      26.400000       0.800000  
max        9.000000      9.000000      40.200000      46.700000     371.000000

### Vemos los tipo de datos de cada columna

In [17]:
data.columns

Index(['Date', 'Location', 'MinTemp', 'MaxTemp', 'Rainfall', 'Evaporation',
       'Sunshine', 'WindGustDir', 'WindGustSpeed', 'WindDir9am', 'WindDir3pm',
       'WindSpeed9am', 'WindSpeed3pm', 'Humidity9am', 'Humidity3pm',
       'Pressure9am', 'Pressure3pm', 'Cloud9am', 'Cloud3pm', 'Temp9am',
       'Temp3pm', 'RainToday', 'RISK_MM', 'RainTomorrow'],
      dtype='object')

In [18]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142193 entries, 0 to 142192
Data columns (total 24 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   Date           142193 non-null  object 
 1   Location       142193 non-null  object 
 2   MinTemp        141556 non-null  float64
 3   MaxTemp        141871 non-null  float64
 4   Rainfall       140787 non-null  float64
 5   Evaporation    81350 non-null   float64
 6   Sunshine       74377 non-null   float64
 7   WindGustDir    132863 non-null  object 
 8   WindGustSpeed  132923 non-null  float64
 9   WindDir9am     132180 non-null  object 
 10  WindDir3pm     138415 non-null  object 
 11  WindSpeed9am   140845 non-null  float64
 12  WindSpeed3pm   139563 non-null  float64
 13  Humidity9am    140419 non-null  float64
 14  Humidity3pm    138583 non-null  float64
 15  Pressure9am    128179 non-null  float64
 16  Pressure3pm    128212 non-null  float64
 17  Cloud9am       88536 non-null

### Buscamos nulos por columnas

In [19]:
data.columns[data.isnull().any()]

Index(['MinTemp', 'MaxTemp', 'Rainfall', 'Evaporation', 'Sunshine',
       'WindGustDir', 'WindGustSpeed', 'WindDir9am', 'WindDir3pm',
       'WindSpeed9am', 'WindSpeed3pm', 'Humidity9am', 'Humidity3pm',
       'Pressure9am', 'Pressure3pm', 'Cloud9am', 'Cloud3pm', 'Temp9am',
       'Temp3pm', 'RainToday'],
      dtype='object')

### - Trabajamos los nulos

<a id = "1"></a><br>
# Descripción de variables

- Date: fecha de la observación.
- Location: ubicación de la estación meteorológica.
- MinTemp: temperatura mínima en grados Celsius.
- MaxTemp: temperatura máxima en grados Celsius.
- Rainfall: cantidad de lluvia registrada ese día en mm.
- Evaporation: evaporación (mm) en 24 horas.
- Sunshine: número de horas de sol brillante en el día.
- WindGustDir: dirección de la ráfaga de viento más fuerte en 24 horas.
- WindGustSpeed: velocidad (km/hr) de la ráfaga de viento más fuerte en 24 horas.
- WindDir9am: dirección del viento a las 9am.
- WindDir3pm: dirección del viento a las 3pm.
- WindSpeed9am: velocidad (km/hr) del viento a las 9am.
- WindSpeed3pm: velocidad (km/hr) del viento a las 3pm.
- Humidity9am: porcentaje de humedad a las 9am.
- Humidity3pm:  porcentaje de humedad a las 3pm.
- Pressure9am: presión atmosférica (hpa) a nivel del mar a las 9am.
- Pressure3pm: presión atmosférica (hpa) a nivel del mar a las 3pm.
- Cloud9am: fración del cielo cubierto por nubes a las 9am. Se mide en "octavos", de manera que un valor 0 indica cielo totalmente despejado y 8, cielo totalmente cubierto.
- Cloud3p fración del cielo cubierto por nubes a las 3pm. Se mide en "octavos", de manera que un valor 0 indica cielo totalmenete 
despejado y 8, cielo totalmente cubieto.:
- Temp: 9temperatura en grados celsius a lasm..m: 
- Temp: 3temperatura en grados celsius a las 3p: 
RainToday:To variable indicadora que toma el valor 1 si la precipitación en mm. en las últimas 24 hrs. excede 1 mm. y 0 si ny:
- RI_MM:SK La cantidad de lluvia. Una especie de medida del "riesgoM:
RainTomorrow:ovariable indicadora que toma el valor 1 si al día siguiente llovió y 0 
si no.no.

<a id = "3"></a><br>
# Análisis de variables

* 17 son variables númericas.
* 7 son variables categóricas.

<a id = "3"></a><br>
# - Variables númericas

<a id = "3"></a><br>
# - Variables categóricas

<a id = "5"></a><br>
# Análisis de datos

### Cruce de variables

In [20]:
# obtener temperatura mínima por ubicación
data[["Location", "MinTemp"]].groupby(["Location"], as_index = False).min().sort_values(by="MinTemp", ascending= True)

Location  MinTemp
23       MountGinini     -8.5
40       Tuggeranong     -8.2
9           Canberra     -8.0
5           Ballarat     -5.6
3       AliceSprings     -5.2
34          Richmond     -4.8
42        WaggaWagga     -4.7
36        SalmonGums     -4.4
6            Bendigo     -4.0
35              Sale     -3.9
17        Launceston     -3.5
28         Nuriootpa     -3.5
21             Moree     -3.3
20           Mildura     -3.2
4      BadgerysCreek     -3.0
2             Albury     -2.8
12          Dartmoor     -2.7
25              Nhil     -2.3
44          Watsonia     -2.1
22      MountGambier     -2.1
10             Cobar     -2.0
41             Uluru     -1.9
29        PearceRAAF     -1.9
45       Williamtown     -1.8
33          Portland     -1.5
30           Penrith     -1.1
32      PerthAirport     -1.0
19  MelbourneAirport     -1.0
46       Witchcliffe     -0.7
31             Perth     -0.6
15            Hobart     -0.4
11      CoffsHarbour      0.6
0           Adelaide      0.7
48           Woomera      0.7
18         Melbourne      1.5
1             Albany      1.8
24         Newcastle      2.2
43           Walpole      2.6
7           Brisbane      2.6
38     SydneyAirport      3.2
37            Sydney      4.3
26         NorahHead      4.6
14         GoldCoast      4.9
16         Katherine      5.2
47        Wollongong      5.4
39        Townsville      6.4
8             Cairns      9.2
27     NorfolkIsland      9.4
13            Darwin     12.5

* La ubicación con la temperatura más baja resgristrada es MountGinini con -8.5 grados.

In [21]:
# obtener temperatura máxima por ubicación
data[["Location", "MaxTemp"]].groupby(["Location"], as_index = False).max().sort_values(by="MaxTemp", ascending= False)

Location  MaxTemp
48           Woomera     48.1
21             Moree     47.3
34          Richmond     47.0
30           Penrith     46.9
19  MelbourneAirport     46.8
44          Watsonia     46.7
20           Mildura     46.7
10             Cobar     46.6
18         Melbourne     46.4
38     SydneyAirport     46.4
4      BadgerysCreek     46.4
36        SalmonGums     46.3
37            Sydney     45.8
0           Adelaide     45.7
45       Williamtown     45.5
29        PearceRAAF     45.4
6            Bendigo     45.4
25              Nhil     45.3
42        WaggaWagga     45.2
3       AliceSprings     44.9
22      MountGambier     44.9
2             Albury     44.8
31             Perth     44.4
12          Dartmoor     44.4
41             Uluru     44.4
35              Sale     44.4
32      PerthAirport     44.2
5           Ballarat     44.1
24         Newcastle     44.1
28         Nuriootpa     44.1
26         NorahHead     44.0
33          Portland     42.5
47        Wollongong     42.4
9           Canberra     42.0
16         Katherine     42.0
15            Hobart     41.8
40       Tuggeranong     41.1
46       Witchcliffe     40.6
43           Walpole     39.4
11      CoffsHarbour     39.2
17        Launceston     39.0
1             Albany     39.0
7           Brisbane     38.9
8             Cairns     38.6
39        Townsville     38.3
13            Darwin     38.0
14         GoldCoast     37.9
23       MountGinini     31.1
27     NorfolkIsland     28.4

* La ubicación con la temperatura más alta es Woomera con 48.1 grados.

In [22]:
# Obtener promedio de cantidad de sol al día por ubicación
data[["Location", "Sunshine"]].groupby(["Location"], as_index = False).mean().sort_values(by="Sunshine", ascending= True)

Location  Sunshine
44          Watsonia  6.378152
19  MelbourneAirport  6.381283
12          Dartmoor  6.412715
18         Melbourne  6.465776
33          Portland  6.496751
22      MountGambier  6.511167
15            Hobart  6.578324
1             Albany  6.658765
35              Sale  6.683663
27     NorfolkIsland  7.021751
45       Williamtown  7.140044
37            Sydney  7.179374
38     SydneyAirport  7.198126
11      CoffsHarbour  7.362374
9           Canberra  7.403241
8             Cairns  7.575995
28         Nuriootpa  7.675888
0           Adelaide  7.752002
7           Brisbane  8.082782
42        WaggaWagga  8.197864
39        Townsville  8.460627
20           Mildura  8.468104
13            Darwin  8.499310
10             Cobar  8.674364
31             Perth  8.784536
32      PerthAirport  8.795306
29        PearceRAAF  8.894160
21             Moree  8.896590
48           Woomera  9.063005
3       AliceSprings  9.581944
2             Albury       NaN
4      BadgerysCreek       NaN
5           Ballarat       NaN
6            Bendigo       NaN
14         GoldCoast       NaN
16         Katherine       NaN
17        Launceston       NaN
23       MountGinini       NaN
24         Newcastle       NaN
25              Nhil       NaN
26         NorahHead       NaN
30           Penrith       NaN
34          Richmond       NaN
36        SalmonGums       NaN
40       Tuggeranong       NaN
41             Uluru       NaN
43           Walpole       NaN
46       Witchcliffe       NaN
47        Wollongong       NaN